src/EditorFeatures/Core/Implementation/IntelliSense/Helpers.cs

```diff
@@ -132,9 +136,11 @@ internal static class Helpers
                 {
                     // This is tagged text getting added to the current line we are building.
                     var style = GetClassifiedTextRunStyle(part.Style);
-                    if (part.NavigationTarget is object &&
-                        context?.ThreadingContext is ThreadingContext threadingContext &&
-                        context.StreamingPresenter is Lazy<IStreamingFindUsagesPresenter> streamingPresenter)
+                    if (part.NavigationTarget is not null &&
+                        context?.ThreadingContext is { } threadingContext &&
```

Comment:
this wasn't introduced in this PR, but I thought of it when I saw all the null checks - it looks like we only pass null for these in the LSP instance (since we can't pass navigation actions).  Maybe instead of the null checks, it would be better to have different types for IntellisenseQuickInfoBuilderContext?  One for LSP which has none of these, and one for local which has them all as non-nullable fields.  That way we could just check that the context is the 'navigation' context.

Only if it doesn't require significant changes to thread through the contexts everywhere.

Reply:
i actually agree with this.  there's a lot of code that passes the OperationExecutor/AsyncOperationListener/ThreadingContext around.  I think we can definitely make that nicer in the future.